### PR TITLE
feat!: add option to automatically login user after registration

### PR DESCRIPTION
**Pull request by nsine**
Targets #891 
And continues this PR https://github.com/accounts-js/accounts/pull/829

## Breaking change
* updates return type of `createUser` mutation to new `CreateUserResult` type

This PR:
* adds autologin feature (can be enabled by setting `enableAutologin` flag to `true` in `AccountServer` config)
* updates graphql-api/client, rest-express/client packages for new `createUser` response type
* adds new `CreateUserResult` type which includes `userId` and `loginResult` fields which is filled by login data in case autologin is enabled and `null` otherwise
* cherry-picks my commit from other branch https://github.com/accounts-js/accounts/pull/894/commits/0e3ec90dad10ccf43f5fd1a0da72d3a2a438ec2c to refactor shared types in graphql-api

### CODE REVIEW

**Comment by nsine** [comment]
@pradel can you please have a look and probably help me with writing tests for this
I tried to quickly understand how tests are organized in the app, but didn't make it 😕

**Comment by pradel** [comment]
@nsine thanks! Will try to take a look before the end of the week

File: packages/graphql-api/src/modules/accounts-password/resolvers/mutation.ts [hunk 1]
```diff
@@ -24,11 +24,36 @@ export const Mutation: MutationResolvers<ModuleContext<AccountsModuleContext>> =
...
+      return {
+        userId: accountsServer.options.ambiguousErrorMessages ? null : userId,
+      };
+    }
+
+    // When initializing AccountsServer we check that enableAutologin and ambiguousErrorMessages options
+    // are not enabled at the same time
+
+    const { password, ...rest } = user;
+    const loginResult = await accountsServer.loginWithService(
```
Comment (pradel) [inline packages/graphql-api/src/modules/accounts-password/resolvers/mutation.ts]: here it's probably better to call `loginWithUser` directly :)

File: packages/rest-express/src/endpoints/password/register.ts [hunk 2]
```diff
@@ -1,15 +1,45 @@
...
+    }
+
+    // When initializing AccountsServer we check that enableAutologin and ambiguousErrorMessages options
+    // are not enabled at the same time
+
+    const userAgent = getUserAgent(req);
+    const ip = requestIp.getClientIp(req);
+
+    const { password, ...rest } = req.body.user;
+    const loginResult = await accountsServer.loginWithService(
```
Comment (pradel) [inline packages/rest-express/src/endpoints/password/register.ts]: here it's probably better to call `loginWithUser` directly :)

**Comment by pradel** [comment]
For the test, first of all not of them have tests. Tests are under an `__tests__` folder, if you run the `yarn coverage` command inside one of the packages, you should see the coverage numbers displayed so just make sure that you add tests for your conditions basically :)

**Comment by nsine** [comment]
@pradel i've added tests for autologin feature and applied your suggestions. Please have a look again 🙂

**Comment by pradel** [comment]
@nsine thank you for the pr, that's a great addition!

**Comment by sakulstra** [comment]
@pradel could we get a release with this :pray:?

**Comment by agustif** [comment]
Awesome just enabled this! Thanks for the hard work !

**Pull request merged**